### PR TITLE
SALTO-6845: (Bugfix) Salesforce `picklistAsMaps` feature does not work for environments with CPQ

### PR DESCRIPTION
**Pull request by tamtamirr**
(Bugfix) Salesforce picklistAsMaps feature does not work for environments with CPQ 

---

For environments that manage CPQ metadata, turning on the `picklistAsMaps` feature makes fetch fail violently. This PR exposes two issues in the adapter:

1. `convert_maps` filter should be able to key-by `ReferenceExpression` value.
2. If `OrderedMap` order value references value with unresolved references, the Element won't be fully resolved.

---
_Release Notes_: 
_None_

---
_User Notifications_: 
_None_

### CODE REVIEW

File: packages/salesforce-adapter/src/adapter.ts [hunk 1]
```diff
@@ -432,6 +436,40 @@ type CreateFiltersRunnerParams = {
   contextOverrides?: Partial<FilterContext>
 }
 
+const isOrderedMapTypeOrRefType = (typeRef: TypeElement | TypeReference): boolean =>
+  typeRef.elemID.name.startsWith('OrderedMap<')
```
Comment (yelly) [inline packages/salesforce-adapter/src/adapter.ts:440]: Didn't you just rename the type? Or is that a different PR?
Either way, look out.

File: packages/salesforce-adapter/src/filters/convert_maps.ts [hunk 2]
```diff
@@ -99,21 +111,22 @@ const createOrderedMapType = <T extends TypeElement>(innerType: T): ObjectType =
...
- */
-export const defaultMapper = (val: string): string[] => val.split(API_NAME_SEPARATOR).map(v => naclCase(v))
+const stringifyMapperInput = (mapperInput: MapperInput): string =>
+  _.isString(mapperInput) ? mapperInput : apiNameSync(mapperInput.value) ?? ''
+
+const isMapperInput = (val: unknown): val is MapperInput =>
+  _.isString(val) || isResolvedReferenceExpressionToElement(val)
+
+export const defaultMapper: MapperFunc = val =>
+  (_.isString(val) ? val : apiNameSync(val.value) ?? '').split(API_NAME_SEPARATOR).map(v => naclCase(v))
```
Comment (yelly) [inline packages/salesforce-adapter/src/filters/convert_maps.ts:121]:
```suggestion
  stringifyMapperInput(val).split(API_NAME_SEPARATOR).map(v => naclCase(v))
```
(I'm assuming this is what you meant to do)

File: packages/salesforce-adapter/src/filters/cpq/lookup_fields.ts [hunk 3]
```diff
@@ -142,16 +143,12 @@ const transformObjectLookupValueSetFullNames = async (
   ) => ReferenceExpression | string | undefined,
 ): Promise<ObjectType> => {
   const lookupFields = await getLookupFields(object)
-  lookupFields.forEach(field => transformLookupValueSetFullNames(field, transformFullNameFn))
+  await awu(lookupFields).forEach(field => transformLookupValueSetFullNames(field, transformFullNameFn))
```
Comment (yelly) [inline packages/salesforce-adapter/src/filters/cpq/lookup_fields.ts:146]: What? How did this work before?

File: packages/salesforce-adapter/src/adapter.ts (hunk 1 above)
Comment (tamtamirr) [reply packages/salesforce-adapter/src/adapter.ts:440]: different PR

File: packages/salesforce-adapter/src/filters/cpq/lookup_fields.ts (hunk 3 above)
Comment (tamtamirr) [reply packages/salesforce-adapter/src/filters/cpq/lookup_fields.ts:146]: race condition. Found this with debugger which is significantly slower.

File: packages/salesforce-adapter/src/filters/convert_maps.ts (hunk 2 above)
Comment (yelly) [reply packages/salesforce-adapter/src/filters/convert_maps.ts]: Did you miss this?

**Review by yelly — APPROVED** [review]

**Pull request merged**